Backpack/BarChart/Classes/BarChartCollectionView.swift

```diff
@@ -0,0 +1,149 @@
+            selectedMarker.trailingAnchor.constraint(equalTo: backgroundView!.trailingAnchor),
+            selectedMarkerBottomConstraint!,
+            selectedMarker.heightAnchor.constraint(equalToConstant: 1.0)
+        ])
+        
+        register(BPKBarChartCollectionViewCell.self,
+                 forCellWithReuseIdentifier: BPKBarChartCollectionView.cellIdentifier)
+
+        collectionViewLayout = layout
+        dataSource = self
+        delegate = self
+        allowsSelection = true
+    }
+
+    lazy fileprivate var layout: UICollectionViewLayout = {
+        let layout = UICollectionViewFlowLayout()
+        layout.estimatedItemSize = CGSize(width: BPKSpacingXxl, height: bounds.height)
+        layout.scrollDirection = .horizontal
+        return layout
+    }()
+
+    lazy fileprivate var selectedMarker: UIView = {
+        let view = UIView()
+        view.backgroundColor = BPKColor.monteverde
+        view.isHidden = true
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    fileprivate func updateSelectedMarkerPosition() {
+        if selectedMarkerBottomConstraint != nil {
+            selectedMarkerBottomConstraint?.isActive = false
+            selectedMarkerBottomConstraint = nil
+        }
+        
+        if selectedIndexPath == nil {
+            selectedMarker.isHidden = true
+        } else {
+            selectedMarker.isHidden = false
+
+            guard let selectedCell: BPKBarChartCollectionViewCell =
+                cellForItem(at: selectedIndexPath!) as? BPKBarChartCollectionViewCell
+            else {
+                return
+            }
+            let selectedBarTopPosition: CGFloat = selectedCell.barChartBar.barTopPosition
+            selectedMarkerBottomConstraint =
+                selectedMarker.bottomAnchor.constraint(equalTo: topAnchor, constant: selectedBarTopPosition)
+            selectedMarkerBottomConstraint?.isActive = true
+        }
+    }
+}
+
+extension BPKBarChartCollectionView: UICollectionViewDataSource {
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return barChartDataSource?.numberOfBarsInChart(barChartCollectionView: self) ?? 0
+    }
+    
+    public func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    public func collectionView(
+        _ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        guard let cell = dequeueReusableCell(
+            withReuseIdentifier: BPKBarChartCollectionView.cellIdentifier,
+            for: indexPath) as? BPKBarChartCollectionViewCell
+            else {
+                fatalError("No cell registered for reuse with identifier \(BPKBarChartCollectionView.cellIdentifier)")
+        }
+        cell.barChartBar.title = barChartDataSource?.titleForBar(barChartCollectionView: self,
```

Comment:
We need to set `cell.isSelected` here too

Reply:
Why? When a cell is created it shouldn't be selected by default 🤔

Reply:
Yeah but what if it is the selected cell? I believe because you aren't setting this here you have issues when scrolling. Keep in my whenever a cell is created you need to ensure that it relfects the underlying data, this includes whether it's selected or not

Reply:
i.e. we want something like `cell.isSelected = selectedIndexPath == indexPath`